src/Sulu/Bundle/CustomUrlBundle/Resources/public/js/components/webspace/settings/custom-url/input/main.js

```diff
@@ -0,0 +1,166 @@
+
+    return {
+
+        defaults: defaults,
+
+        events: {
+            names: {
+                setBaseDomain: {
+                    postFix: 'set-base-domain',
+                    type: 'on'
+                }
+            },
+            namespace: eventNamespace
+        },
+
+        /**
+         * Initializes the component.
+         */
+        initialize: function() {
+            this.render(this.options.baseDomain);
+            this.setDomData(this.$el.data('custom-url-data') || {});
+
+            this.bindDomEvents();
+            this.bindCustomEvents();
+        },
+
+        /**
+         * Bind aura events.
+         */
+        bindCustomEvents: function() {
+            this.events.setBaseDomain(this.setBaseDomain.bind(this));
+        },
+
+        /**
+         * Bind events to dom-elements.
+         */
+        bindDomEvents: function() {
+            this.$el.on('data-changed', function() {
+                this.setBaseDomain(this.$el.data('custom-url-data'));
+            }.bind(this));
+
+            this.$el.on('change', 'input', function() {
+                this.$el.data('custom-url-data', this.getData());
+            }.bind(this));
+        },
+
+        /**
+         * Render base-domain input.
+         *
+         * @param {String} baseDomain
+         */
+        render: function(baseDomain) {
+            this.baseDomain = baseDomain;
+
+            if (this.baseDomain === null) {
+                return;
+            }
+
+            this.html(parseBaseDomain.call(this, this.baseDomain));
+
+            this.$find('.domain-part').each(function() {
+                var text = $(this).text();
+
+                if (text.length > 15) {
+                    $(this).text(util.cropMiddle(text, 15));
+                }
+            });
+        },
+
+        /**
+         * Set base-domain and rerender input with existing data.
+         *
+         * @param {String} baseDomain
+         */
+        setBaseDomain: function(baseDomain) {
+            var data = this.getData();
+            this.render(baseDomain);
+            this.setDomData(data);
+        },
+
+        /**
+         * Returns data from input elements.
+         *
+         * @returns {{prefix: String, suffix: Array}}
+         */
+        getData: function() {
+            var prefix = $('input[data-prefix="true"]').val() || '',
+                suffix = _.map($('input[data-suffix="true"]'), function(item) {
+                    return $(item).val();
+                });
+
+            return {prefix: prefix, suffix: suffix};
+        },
+
+        /**
+         * Set data to dom elements.
+         *
+         * @param {{prefix: String, suffix: Array}} data
+         */
+        setDomData: function(data) {
+            $('input[data-prefix="true"]').val(data.prefix || '');
+            _.map($('input[data-suffix="true"]'), function(item, index) {
+                return $(item).val(data.suffix[index] || '');
```

Comment:
right t isnt used here i will replace it with each which is semantical better than map